coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -570,6 +570,12 @@ private static Schema getUnionFieldSchema(@Nonnull Schema leftSchema, @Nonnull S
           throw new IllegalArgumentException(
               "Unsupported Avro type " + leftSchema.getType() + " in schema: " + leftSchema.toString(true));
       }
+    } else if (leftSchema.getType().equals(BYTES) || rightSchema.getType().equals(BYTES)) {
+      if (leftSchema.getType().equals(FIXED)){
+        return rightSchema;
+      } else if (rightSchema.getType().equals(FIXED)){
+        return leftSchema;
+      }
```

Comment:
I would recommend to simplify this part to be:
```
    } else {
      final ImmutableSet<Schema.Type> types = ImmutableSet.of(leftSchemaType, rightSchemaType);
      if (types.equals(ImmutableSet.of(FIXED, BYTES))) {
        return Schema.create(BYTES);
      }
    }
```
because I feel that we may meet similar issues in the future, this approach can make the code simpler.

Reply:
Sounds good, thanks @ljfgem ! Will update it in next patch